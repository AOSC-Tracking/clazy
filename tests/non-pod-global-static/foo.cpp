#include <QtCore/QCoreApplication>







struct NonPod
{
    NonPod() {}
    ~NonPod() {}
    int a;
};

struct Pod
{
    int a;
};


static Pod p;
static NonPod p2;
static NonPod* p3;
static int p4;

void foo() {}
Q_COREAPP_STARTUP_FUNCTION(foo) // OK, it's blacklisted
