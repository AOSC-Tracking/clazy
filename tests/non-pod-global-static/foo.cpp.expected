non-pod-global-static/foo.cpp:23:1: warning: non-POD static (NonPod) [-Wclazy-non-pod-global-static]
